src/platform_impl/macos/view.rs

```diff
@@ -623,6 +623,18 @@ fn retrieve_keycode(event: id) -> Option<VirtualKeyCode> {
     })
 }
 
+// Update `state.modifiers` if `event` has something different
+fn update_potentially_stale_modifiers(state: &mut ViewState, event: id) {
+    if state.modifiers != event_mods(event) {
+        state.modifiers = event_mods(event);
```

Comment:
I'd suggest assigning `event_mods(event)` to a variable, so it's not executed twice.